packages/dds/tree/src/test/simple-tree/objectNode.spec.ts

```diff
@@ -20,17 +25,83 @@ describeHydration(
 	"ObjectNode",
 	(init) => {
 		describe("shadowing", () => {
-			it("constructor", () => {
-				// constructor is a special case, since one is built in on the derived type.
+			describe("constructor", () => {
+				it("empty", () => {
+					class Schema extends schemaFactory.object("x", {}) {}
+					const n = init(Schema, {});
+					// constructor is a special case, since one is built in on the derived type.
+					// Check that it is exposed as expected based on type:
+					const x = n.constructor;
+					// eslint-disable-next-line @typescript-eslint/ban-types
+					type check_ = requireAssignableTo<typeof x, Function>;
+					assert.equal(x, Schema);
+				});
+
+				it("required", () => {
+					class Schema extends schemaFactory.object("x", {
+						constructor: schemaFactory.number,
+					}) {}
+
+					const n = init(Schema, { constructor: 5 });
+
+					const x = n.constructor;
+					type check_ = requireAssignableTo<typeof x, number>;
+					assert.equal(x, 5);
+				});
+
+				describe("optional", () => {
+					class Schema extends schemaFactory.object("x", {
+						constructor: schemaFactory.optional(schemaFactory.number),
+					}) {}
+
+					it("explicit undefined", () => {
+						const n = init(Schema, { constructor: undefined });
+						const x = n.constructor;
+						type check_ = requireAssignableTo<typeof x, number | undefined>;
+						assert.equal(x, undefined);
+					});
+
+					it("default", () => {
+						// Example of how a type conversion that allows using literals with defaults can still be allowed to compile in the presence of overloaded inherited values.
+						const data: { [P in "constructor"]?: undefined } = {};
+						const insertable: NodeBuilderData<typeof Schema> = data;
+
+						const n = init(Schema, insertable);
+						const x = n.constructor;
+						assert.equal(x, undefined);
+
+						{
+							// In this particular case of overloads, TypeScript knows this is unsafe, but in other similar cases (like the one above).
```

Comment:
Is this sentence unfinished?

Reply:
```suggestion
							// In this particular case of overloads, TypeScript knows this is unsafe, but in other similar cases (like the one above), it can compile without error.
```